File: backend/internal/interface/oapi/geofence.go
package oapi

// TODO 仮定義（open-apiに移行予定）
type GeofenceRequest struct {
	Title       string        `json:"title"`
	UserID      uint64        `json:"userID"`
	Polygon     [][][]float64 `json:"polygon"`
	DeviceToken string        `json:"deviceToken"`
}

File: frontend/src/api/geofence.ts
import { apiClient } from "./client";
import { components } from "./openapi.gen";

export const createGeofence = async (
  polygon: [number, number][][],
  token: string | null
) => {
  console.log("polygon: ", polygon);

  const geofence: components["schemas"]["Geofence"] = {
    title: "hoge title",
    userID: 999,
    polygon: polygon,
    deviceToken: token || "",
  };

  const response = await apiClient.post("/api/geofence", geofence);
};
